### PR TITLE
OPENHELPD-14 FAQ Batch Upload

### PR DESCRIPTION
**Pull request by gsvr**
**DEL12.1**
Create languages from headers (use 6-letter codes)
Labels should be in the last columns (multiple allowed)

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: casepro/msgs/tasks.py [hunk 1]
```diff
@@ -91,3 +103,119 @@ def reply_export(export_id):
...
+        return Language.objects.create(org=org, code=lang_code)
+
+
+def get_labels(labelstring):
+    """
+    Gets a list of label objects from a comma-seperated string of the label codes, eg. "TB, aids"
+    """
+    labels = set()
+    labelstrings = labelstring.split(', ')
+    print labelstrings
```
Comment (imsickofmaps) [inline casepro/msgs/tasks.py]: stray prints (ctrl-f for the rest)

File: casepro/msgs/tasks.py [hunk 2]
```diff
@@ -91,3 +103,119 @@ def reply_export(export_id):
...
+
+
+@task(track_started=True)
+def faq_csv_import(org, task_id):  # pragma: no cover
+    task = ImportTask.objects.get(pk=task_id)
+    log = StringIO()
+
+    task.task_id = faq_csv_import.request.id
+    task.log("Started import at %s" % timezone.now())
+    task.log("--------------------------------")
```
Comment (imsickofmaps) [inline casepro/msgs/tasks.py]: log finish too?

File: casepro/msgs/views.py [hunk 3]
```diff
@@ -26,6 +29,22 @@
...
 
+# Override the ImportTask start method so we can use our self-defined task
+def override_start(self, org):
+    from .tasks import faq_csv_import
+    self.log("Queued import at %s" % now())
+    self.save(update_fields=['import_log'])
+
+    # trigger task
+    result = faq_csv_import.delay(org, self.pk)
+    # TODO: celery setup required? currently always_eager set to true
```
Comment (imsickofmaps) [inline casepro/msgs/views.py]: address or remove?

**Comment by imsickofmaps** [comment]
Needs tests like https://github.com/praekelt/ndoh-control/blob/develop/subscription/tests.py#L169-L204

File: casepro/msgs/views.py (hunk 3 above)
Comment (gsvr) [inline casepro/msgs/views.py]: I'm actually not sure - it didn't look to me like it was necessary to update any settings, but your knowledge of celery is much better than mine - could you have a look if I overlooked something?

File: casepro/msgs/tasks.py (hunk 2 above)
Comment (gsvr) [inline casepro/msgs/tasks.py]: You mean other than this: https://github.com/praekelt/casepro/pull/12/files#diff-96833113def594a421fdef9bac214d27R209 ?

Comment (imsickofmaps) [inline casepro/msgs/tasks.py]: \o/

**Pull request merged**